examples/update-viewState/app.js

```diff
@@ -0,0 +1,225 @@
+  const map = useRef(null);
+
+  const [ready, setReady] = useState(false);
+  const [busy, setBusy] = useState(false);
+  const nextFrame = useNextFrame();
+
+  const [time, setTime] = useState(0);
+  const [animation] = useState({});
+  const [viewState, setViewState] = useState(INITIAL_VIEW_STATE);
+
+  const animate = () => {
+    setTime(t => (t + animationSpeed) % loopLength);
+    animation.id = window.requestAnimationFrame(animate);
+  };
+
+  useEffect(() => {
+    animation.id = window.requestAnimationFrame(animate);
+    return () => window.cancelAnimationFrame(animation.id);
+  }, [animation]);
+
+  const layers = [
+    new PolygonLayer({
+      id: 'ground',
+      data: landCover,
+      getPolygon: f => f,
+      stroked: false,
+      getFillColor: [0, 0, 0, 0]
+    }),
+    new TripsLayer({
+      id: 'trips',
+      data: DATA_URL.TRIPS,
+      getPath: d => d.path,
+      getTimestamps: d => d.timestamps,
+      getColor: d => (d.vendor === 0 ? theme.trailColor0 : theme.trailColor1),
+      opacity: 0.3,
+      widthMinPixels: 2,
+      rounded: true,
+      trailLength: 180,
+      currentTime: time,
+
+      shadowEnabled: false
+    }),
+    new PolygonLayer({
+      id: 'buildings',
+      data: DATA_URL.BUILDINGS,
+      extruded: true,
+      wireframe: false,
+      opacity: 0.5,
+      getPolygon: f => f.polygon,
+      getElevation: f => f.height,
+      getFillColor: theme.buildingColor,
+      material: theme.material
+    })
+  ];
+
+  const updateCamera = (prevCamera) => {
+
+    // Set by User
+  prevCamera = new CameraKeyframes({
+        timings: [0, 5000],
+        keyframes: [
+          {
+            longitude: viewState.longitude,
+            latitude: viewState.latitude,
+            zoom: viewState.zoom,
+            pitch: viewState.pitch,
+            bearing: viewState.bearing
+          },
+          {
+            longitude: viewState.longitude,
+            latitude: viewState.latitude,
+            zoom: viewState.zoom,
+            bearing: viewState.bearing + 192,
+            pitch: viewState.pitch // up to 45/50
+          }
+        ],
+        easings: [easing.easeInOut]
+      });
+   
+    return prevCamera;
+  }
+
+  return (
+    <div style={{position: 'relative'}}>
+      <div style={{position: 'absolute'}}>
+        <ResolutionGuide />
+      </div>
+
+      <DeckGL
+        ref={deckgl}
+        layers={layers}
+        effects={theme.effects}
+
+        controller={true}
+        width={640}
+        height={480}
+        viewState={viewState}
+        onViewStateChange={({viewState}) => {
+          setViewState(viewState);
+        }}
+      
+        {...adapter.getProps(deckgl, setReady, nextFrame)}
+      >
+        <StaticMap
+          ref={map}
+          reuseMaps
+          mapStyle={mapStyle}
+          preventStyleDiffing={true}
+          mapboxApiAccessToken="pk.eyJ1IjoicGlvbmVlci1tZSIsImEiOiJjanA0OXMwM2IwcW5qM2tvYnAyYndpdXMxIn0.bqxGkqM2ozOVT57GuVzEjw"
```

Comment:
Please remove api keys, we can't commit those. The env variables should load these via webpack